config/main.py

```diff
@@ -332,18 +334,20 @@ def interface_name_to_alias(config_db, interface_name):
 
     return None
 
-def interface_ipaddr_dependent_on_interface(config_db, interface_name):
-    """Get table keys including ipaddress
+def get_interface_ipaddresses(config_db, interface_name):
+    """Get IP addresses attached to interface
     """
-    data = []
+    ipaddresses = set()
     table_name = get_interface_table_name(interface_name)
-    if table_name == "":
-        return data
+
     keys = config_db.get_keys(table_name)
     for key in keys:
-        if interface_name in key and len(key) == 2:
-            data.append(key)
-    return data
+        if isinstance(key, tuple) and len(key) == 2:
+            iface, ipaddr = key
```

Comment:
>ipaddr [](http://example.com/codeflow?start=19&length=6)

ipprefix

Reply:
It is not exactly an ipprefix (the CIDR address contains host bits set). Renamed to interface_ip which is a term used everywhere in this code.